File: POO/TP2/Practica Python/EJ10/Comuna.py
from datetime import date

class Comuna:
    __nombre: str
    __familias: list
    __fecha: date
    
    def __init__(self, nombre):
        self.__nombre = nombre
        self.__familias = []
        self.__fecha = date.today

    def get_nombre(self):
        return self.__nombre
    
    def set_nombre(self,nombre):
        self.__nombre = nombre
        
    def get_fecha(self):
        return self.__fecha
    
    def set_fecha(self, fecha):
        self.__fecha = date.fromisoformat(fecha)
    
    def add_familia(self, familia):
        self.__familias.append(familia)
        
    def get_cantidad_familias(self):
        return len(self.__familias)
    
    def get_cantidad_personas(self):
        personas= 0
        for familia in self.__familias:
            personas += len(familia.get_personas())
        return personas
    
    def get_promedio_edad(self):
        suma_edad=0
        personas= 0
        familias = self.__familias
        for familia in familias:
            for persona in familia.get_personas():
                suma_edad += persona.get_edad()
            personas = personas + len(familia.get_personas())
        return suma_edad/personas
    
    def get_trabajadores(self):
        trabajadores=[]
        familias= self.__familias;
        for familia in familias:
            for persona in familia.get_personas():
                if persona.get_trabaja():
                    trabajadores.append(persona)
        return trabajadores
    
    def get_list_pueden_trabajar(self):
        familias = self.__familias
        pueden_trabajar=[]
        for familia in familias:
            for persona in familia.get_personas():
                if persona.get_puede_trabajar():
                    pueden_trabajar.append(persona)
        return pueden_trabajar

    def get_list_pueden_manejar(self):
        familias = self.__familias
        pueden_manejar=[]
        for familia in familias:
            for persona in familia.get_personas():
                if persona.get_puede_manejar():
                    pueden_manejar.append(persona)
        return pueden_manejar

File: POO/TP2/Practica Java/EJ10/Comuna.java
package EJ10;

import java.util.ArrayList;

public class Comuna {
    private String nombre;
    private ArrayList<Familia> familias;

    public Comuna(String nombre) {
        this.nombre = nombre;
        this.familias = new ArrayList<Familia>();
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public ArrayList<Familia> getFamilias() {
        return this.familias;
    }

    public void setFamilias(ArrayList<Familia> familias) {
        this.familias = familias;
    }

    public void addFamilia(Familia familia) {
        this.familias.add(familia);
    }

    public Integer familiasCensadas() {
        return (this.familias).size();
    }

    public Integer getCantidadPersonas() {
        Integer cantidadPersonas = 0;
        for (Familia familia : this.familias) {
            cantidadPersonas += familia.getPersonas().size();
        }
        return cantidadPersonas;
    }

    public Double getPromedioEdad() {
        Integer suma = 0;
        for (Familia familia : this.familias) {
            for (Persona persona : familia.getPersonas()) {
                suma = suma + persona.getEdad();
            }
        }
        double sumaD = suma;
        return (sumaD / this.getCantidadPersonas());
    }

    public Integer getCantidadTrabajadores() {
        Integer trabajadores = 0;
        for (Familia familia : this.familias) {
            for (Persona persona : familia.getPersonas()) {
                if (persona.getTrabaja()) {
                    trabajadores = trabajadores + 1;
                }
            }
        }
        return trabajadores;
    }
}
